demoji/__init__.py

```diff
@@ -217,3 +217,42 @@ def replace(string, repl=""):
     """
     set_emoji_pattern()
     return _EMOJI_PAT.sub(repl, string)
+
+
+def findall_list(string):
+    """Find emojis within ``string`` like findall() but returns a list with each occurrence instead of dictionary. 
+    Useful for looping through. Added by David Agudelo (@davidagud)
+
+    :param string: The input text to search
+    :type string: str
+    :return: A list of ``[emoji: description]``
+    :rtype: list
+    """
+    
+    set_emoji_pattern()
+
+    return [_CODE_TO_DESC[f] for f in list(_EMOJI_PAT.findall(string))]
+
+
+def replace_w_desc(string, separator=":"):
+    """Replace emojis in ``string`` with their description and a ``separator`` immediately before and after ``string.
+    Added by David Agudelo (@davidagud)
+
+    :param string: The input text to search
+    :type string: str
+    :param separator: String put before and after the emoji description
+    :type separator: str
+    :return: Modified ``str`` with replacements made and separator immediately before and after
+    :rtype: str
+    """
+
+    set_emoji_pattern()
+
+    emoji_dict = findall(string)
+
+    updated_string = string
+
+    for emoji in emoji_dict:
+        updated_string = string.replace(emoji, separator + emoji_dict[emoji] + separator)
```

Comment:
This will only provide one updated to `updated_string`, since it is being re-defined with each iteration while `string` is not.